doc/application/coccinelle.rst

```diff
@@ -0,0 +1,510 @@
+the Coccinelle files and coccicheck have been updated.
+
+Coccinelle is available through the package manager
+of many distributions, e.g. :
+
+.. hlist::
+   * Debian
+   * Fedora
+   * Ubuntu
+   * OpenSUSE
+   * Arch Linux
+   * NetBSD
+   * FreeBSD
+
+Some distribution packages are obsolete and it is recommended
+to use the latest version released from the Coccinelle homepage at
+http://coccinelle.lip6.fr/
+
+Or from Github at:
+
+https://github.com/coccinelle/coccinelle
+
+Once you have it, run the following commands:
+
+.. code-block:: console
+
+        ./autogen
+        ./configure
+        make
+
+as a regular user, and install it with:
+
+.. code-block:: console
+
+        sudo make install
+
+More detailed installation instructions to build from source can be
+found at:
+
+https://github.com/coccinelle/coccinelle/blob/master/install.txt
+
+Supplemental documentation
+**************************
+
+For Semantic Patch Language(SmPL) grammar documentation refer to:
+
+http://coccinelle.lip6.fr/documentation.php
+
+Using Coccinelle on Zephyr
+**************************
+
+``coccicheck`` checker is the front-end to the Coccinelle infrastructure
+and has various modes:
+
+Four basic modes are defined: ``patch``, ``report``, ``context``, and
+``org``. The mode to use is specified by setting the MODE variable with
+``--mode=<mode>`` or ``-m=<mode>``.
+
+* ``patch`` proposes a fix, when possible.
+
+* ``report`` generates a list in the following format:
+  file:line:column-column: message
+
+* ``context`` highlights lines of interest and their context in a
+  diff-like style.Lines of interest are indicated with ``-``.
+
+* ``org`` generates a report in the Org mode format of Emacs.
+
+Note that not all semantic patches implement all modes. For easy use
+of Coccinelle, the default mode is ``report``.
+
+Two other modes provide some common combinations of these modes.
+
+- ``chain`` tries the previous modes in the order above until one succeeds.
+
+- ``rep+ctxt`` runs successively the report mode and the context mode.
+  It should be used with the C option (described later)
+  which checks the code on a file basis.
+
+Examples
+********
+
+To make a report for every semantic patch, run the following command:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=report
+
+To produce patches, run:
+
+.. code-block:: console
+
+        ./scripts/coccicheck --mode=patch
+
+The coccicheck target applies every semantic patch available in the
+sub-directories of ``scripts/coccinelle`` to the entire source code tree.
+
+For each semantic patch, a commit message is proposed.  It gives a
+description of the problem being checked by the semantic patch, and
+includes a reference to Coccinelle.
+
+As any static code analyzer, Coccinelle produces false
+positives. Thus, reports must be carefully checked, and patches
+reviewed.
+
+To enable verbose messages set --verbose=1 option, for example:
```

Comment:
Check for consistent use of code block role shorthand for options. Sometimes short commands are using the double-backtick syntax and other times you're not.  For example, this line should say 

    set ``--verbose=1`` option